app/models/user.rb

```diff
@@ -1,2 +1,17 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_counter
+    update(posts_counter: posts.count)
+  end
+
+  def update_comments_counter
+    update(comments_counter: comments.count)
+  end
```

Comment:
- [ ] The user model does not have a comments_counter column so there is no use in having this method here. Kindly remove it.